helix-term/src/health.rs

```diff
@@ -291,6 +308,29 @@ pub fn language(lang_str: String) -> std::io::Result<()> {
     Ok(())
 }
 
+/// Display diagnostics about multiple LSPs and DAPs.
+fn probe_protocols(protocol_name: &str, server_cmds: &[&str]) -> std::io::Result<()> {
```

Comment:
The collect and `as_slice` calls in the caller can be removed if we have this accept an iterator:

```rust
fn probe_protocols<'a, I: Iterator<Item = &'a str> + 'a>(
    protocol_name: &str,
    server_cmds: I,
) -> std::io::Result<()> {
```